File: ruby/types/type_constructors.rb
#!/usr/bin/ruby2.7

require 'ostruct'

i = Integer(5)
f = Float(2.4)
x = Complex(1, 2)
s = String("hello") # or String.new("hello")
r = Range.new(3, 7)
a = Array.new(3) { |i| 2 * i + 3 }
h = Hash.new { |hash, key| {'2' => "Hello, there!", 'color' => 'orange', '3.14' => 'PI'}[key] }
u = Proc.new {|number| number * 3 }
o = OpenStruct.new({
  '2' => "Hello, there!",
  'color' => 'orange',
  '3.14' => 'PI',
})
co = Class.new do
  def number
    @number
  end
  def set_number(number)
    @number = number
  end
  def get_number
    number
  end
end.new

puts "i = Integer(5) # integer: #{i} (#{i.class})\n\n"
puts "f = Float(2.4) # floating point: #{f} (#{f.class})\n\n"
puts "x = Complex(1, 2) # complex: #{x} (x.class)\n\n"
puts "s = String(\"hello\") # string: #{s} (#{s.class})\n\n"
puts "r = Range.new(3, 7) # range: #{r} (#{r.class})\n\n"
puts "a = Array.new(3) { |i| 2 * i + 3 } # array: #{a} (#{a.class})\n\n"
puts "h = Hash.new { |hash, key| {'2' => \"Hello, there!\", 'color' => 'orange', '3.14' => 'PI'}[key] } # hash: #{h} (#{h.class})\n\n"
puts "u = Proc.new {|number| number * 3 } # function: #{u} (#{u.class})\n\n"
puts "o = OpenStruct.new({\n  '2' => \"Hello, there!\",\n  'color' => 'orange',\n  '3.14' => 'PI',\n})\n# object: #{o} (#{o.class})\n\n"
puts "co = Class.new do\n  def number\n    @number\n  end
  def set_number(number)\n    @number = number\n  end
  def get_number\n    number\n  end
end.new
# object: #{co} (#{co.class})\n\n"

File: python/types/type_constructors.py
#!/usr/bin/python3

b = bool(True)
i = int(5)
f = float(2.4)
x = complex(1+2j)
s = str('hello')
r = range(3, 7, 2)
l = list((3, 5, 7))
t = tuple((3, 5, 7))
e = set((3, 5, 7))
z = frozenset((3, 5, 7))
g = dict(
  two = "Hello, there!",
  color = 'orange',
  three_point_fourteen = 'PI',
)
y = bytes(5)
j = bytearray(3)
m = memoryview(bytes(5))

print(f"b = bool(True) # boolean: {b} ({type(b)})\n")
print(f"i = int(5) # integer: {i} ({type(i)})\n")
print(f"f = float(2.4) # floating point: {f} ({type(f)})\n")
print(f"x = conplex(1+2j) # complex: {x} ({type(x)})\n")
print(f"s = str(\"hello\") # string: {s} ({type(s)})\n")
print(f"r = range(3, 7, 2) # range: {r} ({type(r)})\n")
print(f"l = list((3, 5, 7)) # list: {l} ({type(l)})\n")
print(f"t = tuple((3, 5, 7)) # tuple: {t} ({type(t)})\n")
print(f"e = set((3, 4, 5)) # set: {e} ({type(e)})\n")
print(f"z = frozenset((3, 4, 5)) # frozen set: {z} ({type(z)})\n")
print(f"g = dict(\n  two = \"Hello, there!\",\n  color = 'orange',\n  three_point_fourteen = 'PI',\n)\n# dictionary: {g} ({type(g)})\n")
print(f"y = bytes(5) # bytes: {y} ({type(y)})\n")
print(f"j = bytearray(3) # byte array: {j} ({type(j)})\n")
print(f"m = memoryview(bytes(5)) # memory view: {m} ({type(m)})\n")
